HW06.Rmd

```diff
@@ -1,92 +1,128 @@
+
+GBM.transcripts <- bind_cols(c(gene_names, lapply(files_to_open, open_files_for_TPM)))
+```
+
+Now, install the packages commented below (if needed), and then use this code to map the transcript IDs to gene symbols. To use this code, you need a dataframe called GBM.transcripts that has the first column "gene_id" that contains the transcript ids (e.g. ENST00000456328.2) and the remaining columns contain the TPM data. So long as the first column contains the "gene_id" column as mentioned above, this *should* run. 
+
+```{r map gene.ids to gene.symbols using ensembldb, message = FALSE}
+#install.packages("BiocManager")
+#BiocManager::install("ensembldb")
+#BiocManager::install("EnsDb.Hsapiens.v75")
+library(ensembldb)
+library(EnsDb.Hsapiens.v75)
+
+ens.GBM.transcripts <- GBM.transcripts %>% 
+  mutate(gene_id = gsub(pattern = "\\..*", "", .$gene_id))
+
+map <- ensembldb::select(EnsDb.Hsapiens.v75, keys = ens.GBM.transcripts$gene_id,
+                         keytype = "TXID", columns = c("SYMBOL", "TXID"))
+
+ens.mapped_GBM <- left_join(ens.GBM.transcripts, map, by = c("gene_id" = "TXID")) %>% 
+  dplyr::select(-1) %>% 
+  dplyr::select(gene_symbol = SYMBOL, everything())
+
+ens.mapped_GBM <- ens.mapped_GBM[!duplicated(ens.mapped_GBM$gene_symbol),] #remove duplicated gene symbols
+  #these are removed instead of averaged because they simply do not correlate particularly well. 
+ens.mapped_GBM <- ens.mapped_GBM[!is.na(ens.mapped_GBM$gene_symbol),] #remove NA values
+```
+
+Do the t-test and make a table of the t-test results!
+
+```{r t.test for difference between a bunch of genes}
+#run this code to unload the libraries from before, it might be helpful because the select() function from dplyr might be hidden otherwise
+detach(package:EnsDb.Hsapiens.v75, unload = T)
+detach(package:ensembldb, unload = T)
+
+#add in your own gene of interest!!! 
+#I added 5S_rRNA :)
+genes_of_interest <- c("IRX1", "OSR1", "DCHS2", "BRINP3", "TOB2P1", "FOXD1", "ZFPM2", "GLB1", "ALG5", "TRIM4", "ADARB2", "PCDHGA11", "IDH1", "EGFR", "MGMT", "TERT", "PTEN", "TP53", "RB1", "PTEN", "ATRX", "PDGFRA", "PIK3CA", "MICA", "CDKN2A", "EZH2", "BRD2", "5S_rRNA")
+
+GBM.genes.of.interest <- filter(ens.mapped_GBM, gene_symbol %in% genes_of_interest)
+
+#Now perform a t-test between the H3K mutated and the wt samples. There are many ways to do this actually, you can use a for loop or you could do the tidy alternative with broom(), but the for loop is probably the easiest
+
+#to work in the tidyverse, it will be easier to make tidy the dataframe first
+
+tidy.GBM.genes.of.interest <- GBM.genes.of.interest %>% 
+  gather(key = "key", value = "value", -gene_symbol)  %>%
+  separate(key, into = c ("group", "order"), convert = TRUE)
+
+for (gene in genes_of_interest) {
+  gene_df <- tidy.GBM.genes.of.interest  %>%
+    filter(gene_symbol == gene)
+
+  #print out the t-test results
+  print(paste("========", gene, "========"))
+  print(t.test(value ~ group, gene_df))
+}
+```
```

Comment:
Good use of tideverse to tidy the dataframe and very clean execution of for loop and t.test function to perform the test in one go. I wasn't aware of the use of the ~ operator to specify the groups to compare without explicitly calling them.